subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1127,6 +1127,98 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+static int cmd_wifi_promisc_mode(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_promisc_setup promisc_mode = {0};
+	int ret;
+	long mode;
+
+	if (argc == 1) {
+		promisc_mode.oper = WIFI_MGMT_GET;
+	} else if (argc == 2) {
+		promisc_mode.oper = WIFI_MGMT_SET;
+		if (!parse_number(sh, &mode, argv[1], WIFI_PROMISC_DISABLE,
+				WIFI_PROMISC_STA_AP_MODE)) {
+			return -EINVAL;
+		}
+		promisc_mode.mode =  mode;
+	} else {
+		return -EINVAL;
+	}
+
+	promisc_mode.op = WIFI_PROMISC_MODE;
+	ret = net_mgmt(NET_REQUEST_WIFI_PROMISC_SETUP, iface,
+			&promisc_mode, sizeof(promisc_mode));
+	if (ret) {
+		shell_fprintf(sh, SHELL_ERROR, "Unable to perform %s promisc mode operation: %d\n",
+				promisc_mode.oper == WIFI_MGMT_GET ? "get" : "set", ret);
+		return -ENOEXEC;
+	}
+
+	if (promisc_mode.oper == WIFI_MGMT_GET) {
+		shell_fprintf(sh, SHELL_NORMAL, "Wi-Fi current Promisc mode is %d\n",
+				promisc_mode.mode);
```

Comment:
I think it'd make sense to introduce `wifi_promiscuous_mode2str()` helper function, as for other types, and report the mode as a string, it should be easier to read the shell output that way.

Reply:
will provide a helper function